test/test_tensordict.py

```diff
@@ -1115,6 +1115,159 @@ def test_items_values_keys(self, td_name):
         items_meta = list(td.items_meta())
         assert len(values_meta) == len(items_meta)
 
+    @pytest.mark.parametrize(
+        "td_name",
+        [
+            "td",
+            "stacked_td",
+            "sub_td",
+            "idx_td",
+            "saved_td",
+            "unsqueezed_td",
+            "td_reset_bs",
+        ],
+    )
+    def test_set_requires_grad(self, td_name):
+        td = getattr(self, td_name)
+        assert not td._get_meta("a").requires_grad
+        td.set("a", torch.randn_like(td.get("a")).requires_grad_())
+        assert td._get_meta("a").requires_grad
+
+
+class TestTensorDictsRequiresGrad:
```

Comment:
oh wow that's a whole new class!
I'm ok with it, great work

Reply:
It helped to fund couple bugs, totally worth it

Reply:
indeed!